### DISCUSSION TITLE
NotSignedIn - 2 accounts in VS 2022

### DISCUSSION THREAD

**Discussion by szyb** [community/community#14322] [Copilot Conversations]
Hi,

I have 2 GitHub accounts in VS 2022. The first one is not signed in, but the second is. The problem is that Copilot does not recognize the second account as valid and signed in. Is there any solution to use Copilot extension?

![image](https://user-images.githubusercontent.com/1598139/161748020-20d1a715-0f43-4736-b844-d7f60160f422.png)
Another problem is that I cannot remove the first one. This issue was already sent to VS team.

**Comment by kzu** [comment]
GitHub Copilot does not use the accounts from that dialog. It uses a [device flow](https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps#device-flow) for authorization, which is completely independent from those accounts in VS. 

If you click on the Copilot icon right next to the zoom level in the editor, you will see a menu that allows you to Sign In. At that point, a device code will be copied to the clipboard (if possible) and the default browser will navigate to https://github.com/login/device. Note that at this point whichever GitHub account is signed in in that browser, is the one that will be authorizing the "device" (Visual Studio 2022). If you want to switch accounts, you can open an incognito window, navigate to the same page, sign-in with the desired account, and paste the authorization code there.

**Reply by szyb** [reply]
Thanks, I didn't know that. However it didn't help. When I click Sign In option there is a message box shown and then another "NotSignedIn" message in output. There were no navigation to any login page. Perhaps there is a problem with getting a device code.

![image](https://user-images.githubusercontent.com/1598139/161911513-221b1958-4a16-4a3a-adef-2314156e8ee4.png)

**Reply by szyb** [reply]
Hm... how about proxy support? I'm behind of proxy. If it uses system settings then it should be just fine. I turned VS in verbose mode and got the logs:
```
LSPClient.Copilot Information: 7 : {"id":5,"error":{"code":-32603,"message":"Request signInInitiate failed with message: getaddrinfo ENOTFOUND 4481"}}
LSPClient.Copilot Verbose: 7 : Received: {
  "jsonrpc": "2.0",
  "id": 5,
  "error": {
    "code": -32603,
    "message": "Request signInInitiate failed with message: getaddrinfo ENOTFOUND 4481"
  }
}
```

where 4481 is the proxy port.

I wonder where Copilot wants to connect? Is this anything other than standard ports 80/443? If yes, then my proxy will not pass through such requests.

**Reply by kzu** [reply]
Ah, yes, (proper lack of) proxy support is a known issue, I'm afraid.

**Reply by OriLiMu** [reply]
> Hm... how about proxy support? I'm behind of proxy. If it uses system settings then it should be just fine. I turned VS in verbose mode and got the logs:
> 
> ```
> LSPClient.Copilot Information: 7 : {"id":5,"error":{"code":-32603,"message":"Request signInInitiate failed with message: getaddrinfo ENOTFOUND 4481"}}
> LSPClient.Copilot Verbose: 7 : Received: {
>   "jsonrpc": "2.0",
>   "id": 5,
>   "error": {
>     "code": -32603,
>     "message": "Request signInInitiate failed with message: getaddrinfo ENOTFOUND 4481"
>   }
> }
> ```
> 
> where 4481 is the proxy port.
> 
> I wonder where Copilot wants to connect? Is this anything other than standard ports 80/443? If yes, then my proxy will not pass through such requests.

I get the same question how to turn vs in verbose mode, and is there a way to change the port of the copilot in Visual Studio ?

**Reply by PisChiao** [reply]
I have the same question

**Reply by M4tthiasStahl** [reply]
I have the same question, is there meanwhile an answer?
